src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -42,21 +53,29 @@ function getFilteredWorkspaceList(
   return [
     ...(currentWorkspace ? [currentWorkspace] : []),
     ...workspaceList.filter((workspace) => workspace.id !== currentWorkspace?.id),
-  ].slice(0, 5);
+  ].slice(0, 7);
 }
 
 export const WorkspaceMenu = ({ coreStart }: Props) => {
   const [isPopoverOpen, setPopover] = useState(false);
   const currentWorkspace = useObservable(coreStart.workspaces.currentWorkspace$, null);
   const workspaceList = useObservable(coreStart.workspaces.workspaceList$, []);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const useCasePanels: EuiContextMenuPanelDescriptor[] = [];
 
   const defaultHeaderName = i18n.translate(
     'core.ui.primaryNav.workspacePickerMenu.defaultHeaderName',
     {
       defaultMessage: 'Select a workspace',
     }
   );
-  const filteredWorkspaceList = getFilteredWorkspaceList(workspaceList, currentWorkspace);
+
+  const filteredWorkspaceList = useMemo(() => {
+    return getFilteredWorkspaceList(workspaceList, currentWorkspace).filter((workspace) =>
+      workspace.name.toLowerCase().includes(searchValue.toLowerCase())
+    );
```

Comment:
I think we should do filter first(search), then call `getFilteredWorkspaceList` to limit the maximum number of workspace to display in the dropdown menu.

Reply:
You mean that user can search for the workspaces which is not shown in the picker？